pkg/lib/declarative/indexconfig.go

```diff
@@ -0,0 +1,114 @@
+package declarative
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+	"github.com/operator-framework/operator-registry/pkg/image"
+	"github.com/operator-framework/operator-registry/pkg/image/containerdregistry"
+	"github.com/operator-framework/operator-registry/pkg/image/execregistry"
+	"github.com/operator-framework/operator-registry/pkg/lib/certs"
+	libregistry "github.com/operator-framework/operator-registry/pkg/lib/registry"
+	registry "github.com/operator-framework/operator-registry/pkg/registry"
+
+	"github.com/sirupsen/logrus"
+)
+
+// ConfigEditor is an implementation of IndexConfig
+type ConfigEditor struct {
+	Logger *logrus.Entry
+}
+
+// AddToConfig persists the bundle in it's package config
+func (c ConfigEditor) AddToConfig(request AddConfigRequest) error {
```

Comment:
Ref: https://github.com/operator-framework/enhancements/blob/master/enhancements/declarative-index-config.md#migration-plan

This will be the ONLY way to add bundle in the new binary release.

Reply:
As a side-effect of having a config file as input, I'd imagine it's possible to manually add a bundle to a config. I'm basically wondering if _reading_ a config and serving its content are a better place to start than adding to an existing config.